chain/client/src/client.rs

```diff
@@ -2711,3 +2751,20 @@ impl Client {
         Ok(ret)
     }
 }
+
+impl Drop for Client {
+    fn drop(&mut self) {
+        // Stop the running actors.
+        let epoch_id = self.chain.store().head().expect("Cannot get chain head.").epoch_id;
+        let shard_layout =
+            self.epoch_manager.get_shard_layout(&epoch_id).expect("Cannot get shard layout.");
```

Comment:
Using `epoch_manager` during `drop()` seems complicated.
How about asking `state_sync_adapter` to drop all handles it owns?